qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -0,0 +1,279 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from itertools import zip_longest
+from os.path import basename, join
+
+from qiita_client import ArtifactInfo
+
+from qiita_client.util import system_call, get_sample_names_by_run_prefix
+
+
+def make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file):
+    """Recovers read pairing information
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+
+    Returns
+    -------
+    samples: list of tup
+        list of 3-tuples with run prefix, sample name, fwd read fp, rev read fp
```

Comment:
Good idea but I think that change will change the behavior compared to other plugins, which points that perhaps we should have a method to do this in qiita_client vs. having ad hoc functions. What do you think?